Make generated config attributes internal

gtrepta opened the pull request:
More stuff for #3481 . The `CheckAtt` step happens after the definition has been populated with sentences for the configuration, so some of the attributes it's checking weren't actually added by the user. Changing these attributes to internal will make some checks I'm adding for attribute/sentence restrictions a little cleaner.

There are `cell` and `internal` attributes used here in the `RULE-CELLS` module in `kast.md` which could be addressed later.
https://github.com/runtimeverification/k/blob/3d7841ae51afe44a64931a0692be5c41d6c7e67c/k-distribution/include/kframework/builtin/kast.md?plain=1#L287-L295

gtrepta committed:
Make generated configuration attributes internal

gtrepta committed:
Remove unused initial attribute

dwightguth commented:
I think the reason why these attributes weren't marked as internal is because it is intended that a configuration declaration is identical in semantics to the same set of productions and rules that it generates. That requires users to be able to write the productions that it corresponds to themselves.

dwightguth commented:
Also, the initial attribute isn't dead. It controls how the initializer for cells with multiplicity attribute is generated.

dwightguth reviewed (changes requested):
Marking this as requesting changes. See my comments above. I don't want to just close this PR on you, but I don't think this particular set of changes makes sense

gtrepta commented:
@dwightguth 

> a configuration declaration is identical in semantics to the same set of productions and rules that it generates. That requires users to be able to write the productions that it corresponds to themselves.

Why that requirement?

> Also, the initial attribute isn't dead. It controls how the initializer for cells with multiplicity attribute is generated.

Can you show me an example where the if block I deleted actually gets reached? It was introduced in a commit that added the ocaml backend, and it was never covered when I changed it to just throw an exception.

gtrepta commented:
Closing this for now.

dwightguth commented:
@gtrepta one reason why we might want to do what I described is if we ever want to parse a k definition, serialize the parsed definition to disk, then reload and finish compiling it later. I could very easily see pyk choosing to do this at some point in the future.

I'm not entirely sure where an example might exist, but essentially, if you have a cell of multiplicity `?`, the attribute controls whether the cell starts out initialized or starts out empty.
